src/particle.cpp

```diff
@@ -197,6 +197,12 @@ void Particle::event_advance()
   // Select smaller of the two distances
   double distance = std::min(boundary().distance, collision_distance());
 
+  // Kill particle if its time exceeds the cutoff
+  if (time() + (distance / this->speed()) >
```

Comment:
Why not just do the check after updating the time variable, rather than doing this extra arithmetic here?